### PR TITLE
feat: optimizing deployment framework

### PR DESCRIPTION
**Pull request by drduhe**
**Issue #, if available:** n/a

### Notes
This PR introduces significant refactoring and enhancements to the `osml-cdk-constructs` package to support a more streamlined and maintainable configuration and deployment process. Key changes include:

- **Introduction of `BaseConfig` Class**: A new `BaseConfig` class has been implemented to serve as a foundation for all configuration classes, promoting code reuse and reducing redundancy.
- **Creation of Configuration Classes**: Specific configuration classes (`MESMEndpointConfig`, `OSMLTestImageryConfig`, `MEContainerConfig`, etc.) have been added, encapsulating properties and default values, thereby simplifying component setup.
- **Centralization of Context Values**: Configuration values are now managed via `context.json`, allowing for centralized, consistent context handling across all components.
- **Consolidation of Resources**: Resources for each component have been condensed into single stacks, making the deployment process more straightforward and reducing complexity.
- **Reusable Constructs**: Introduced shared constructs, such as `OSMLContainer`, to handle common tasks like container deployment to ECR, ensuring consistency and reducing code duplication.
- **Extended TSDoc Comments**: Updated TSDoc comments to be more comprehensive and normalized across the package. 
- **Removed TS DLQ**: Updated the ``OSMLTileServerDataplane`` to remove the DLQ table update lambda as it is no longer needed by the application.

These changes collectively enhance the maintainability, readability, and scalability of the `osml-cdk-constructs` package, facilitating a more efficient development and deployment process.

### Checklist

Before you submit a pull request, please make sure you have the following:
- [x] Code changes are compact and well-structured to facilitate easy review
- [x] Changes are documented in the README.md and other relevant documentation pages
- [x] PR title and description accurately reflect the changes and are detailed enough for historical tracking
- [x] PR contains tests that cover all new code and the code has been manual tested
- [x] All new dependencies are declared (if any), and no unnecessary libraries are added
- [x] Performance impacts (if any) of the changes are evaluated and documented
- [x] Security implications of the changes (if any) are reviewed and addressed
- [x] I have read the [Contributing Guidelines](../CONTRIBUTING.md) and agree to follow the [Code of Conduct](../CODE_OF_CONDUCT.md)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: lib/osml/data_catalog/dc_dataplane.ts [hunk 1]
```diff
@@ -19,57 +18,210 @@ import { LambdaSubscription } from "aws-cdk-lib/aws-sns-subscriptions";
...
+  public LAMBDA_MEMORY_SIZE: number;
+
+  /**
+   * The storage size of the Lambda function.
+   * @default 10
+   */
+  public LAMBDA_STORAGE_SIZE: number;
+
+  /**
+   * The timeout of the Lambda function.
```
Comment (natekruse-aws) [inline lib/osml/data_catalog/dc_dataplane.ts:51]: nit: unit again :)

File: lib/osml/utils/regional_config.ts
Comment (natekruse-aws) [inline lib/osml/utils/regional_config.ts:1]: <3

**Review by natekruse-aws — COMMENTED** [review]
I like the overall refactor. Hard to spot check everything. Will give it another look when you post the final revision and all of the actions pass.

File: lib/osml/data_catalog/dc_dataplane.ts
[the remark this answers is not in the document]
Comment (drduhe) [reply lib/osml/data_catalog/dc_dataplane.ts:1]: Ping me offline to discuss

**Review by jtblack-aws — DISMISSED** [review]

File: package.json [hunk 2]
```diff
@@ -15,32 +15,32 @@
...
+    "@types/node": "^22.0.0",
+    "@typescript-eslint/eslint-plugin": "^7.18.0",
+    "@typescript-eslint/parser": "^7.18.0",
+    "aws-cdk": "^2.150.0",
+    "aws-cdk-lib": "^2.150.0",
     "cdk-ecr-deployment": "^3.0.67",
-    "cdk-nag": "^2.28.117",
+    "cdk-nag": "^2.28.168",
     "constructs": "^10.3.0",
     "eslint": "^8.56.0",
```
Comment (jtblack-aws) [inline package.json:29]: probably a future task but I think we can upgrade eslint too.

**Review by jtblack-aws — APPROVED** [review]

**Pull request merged**